### PR TITLE
Remove passing of explicit generator plot_kwargs

### PR DESCRIPTION
**Pull request by cormensratio**
Removes the explicit pass of `plot_kwargs` to plots within their corresponding generator, but passes all `plot_kwargs` instead.
Affected plots/generators: `interaction-degree-plot`, `sankey-plot-rev`, `sankey-plot-cs`, `graphviz-plot-rev`, `graphviz-plot-cs`, `cs-overview-plot`, `pc-overview-plot`

### CODE REVIEW

File: varats/varats/plots/blame_interaction_degree.py [hunk 1]
```diff
@@ -1652,22 +1637,17 @@ class GraphvizLibraryInteractionsGeneratorRev(
...
 
     def generate(self) -> tp.List[Plot]:
+        c_map: CommitMap = get_commit_map(
+            self.plot_kwargs["case_study"].project_name
+        )
+        self.plot_kwargs.update(
+            revision=c_map.convert_to_full_or_warn(
+                ShortCommitHash(self.plot_kwargs["revision"])
+            )
+        )
```
Comment (boehmseb) [inline varats/varats/plots/blame_interaction_degree.py]: Why use `update()` here and not simply `self.plot_kwargs[revision] = cmap.convert_to_full_or_warn(...)`?

**Review by boehmseb — CHANGES_REQUESTED** [review]

File: varats/varats/plots/blame_interaction_degree.py (hunk 1 above)
Comment (cormensratio) [reply varats/varats/plots/blame_interaction_degree.py]: Does it make a difference semantically or is `update()` just not as readable?

Comment (boehmseb) [reply varats/varats/plots/blame_interaction_degree.py]: I'd say that `update()` is less readable. It is usually used to update multiple values at once. But here, you only set a single value.

Comment (cormensratio) [reply varats/varats/plots/blame_interaction_degree.py]: Thanks for the clarification, I'll change it accordingly.

**Review by boehmseb — APPROVED** [review]

**Review by vulder — APPROVED** [review]

**Pull request merged**